assets/src/scripts/components/graph/view/axes.js

```diff
@@ -1,39 +1,60 @@
 import { axisBottom, axisLeft } from 'd3-axis';
 import { timeFormat } from 'd3-time-format';
 
-
+import { FOCUS_CIRCLE_RADIUS } from './cursor';
 
 /**
  * Draws the x-axis
- * @param  {Object} svg              D3 selector
+ * @param  {Object} elem             D3 selector
  * @param  {Function} options.xScale D3 scale function
  * @param  {Object} options.layout   {width, height}
  */
-export const drawAxisX = function (svg, {xScale, layout}) {
-    svg.selectAll('.x-axis').remove();
-    svg.append('g')
-        .classed('x-axis', true)
+export const drawAxisX = function (elem, {xScale, layout}, axis) {
+    axis = axis || elem
+        .append('g')
+        .classed('x-axis', true);
+    axis.transition().duration(100)
         .attr('transform', `translate(0, ${layout.height})`)
         .call(axisBottom()
             .scale(xScale)
             .tickSizeOuter(0)
             .tickFormat(timeFormat('%b %d, %Y')));
+    return axis;
```

Comment:
Why is the axis being returned?

Reply:
This is using the `context` parameter of `link`... The return value from the function will be passed in as the third argument on the next invocation. Here, it's reusing the `g` element created the first time the function was called. See here: https://github.com/ACWI-SOGW/ngwmn-ui/blob/master/assets/src/scripts/lib/d3-redux.js#L50

I've found this pattern to be more useful than the `initAndUpdate` function, because it keeps the create and update logic in the same place. It's also useful to make sure the DOM node for the axis always has the same z-index relative to neighboring nodes.

Reply:
👍